### PR TITLE
FALCON-2146 EntitySlaAlertService runs only once irrespective of the …

### PR DESCRIPTION
**Pull request by PraveenAdlakha**
…configured frequency

### CODE REVIEW

File: client/src/main/java/org/apache/falcon/entity/v0/EntityType.java [hunk 1]
```diff
@@ -124,4 +128,27 @@ public static EntityType getEnum(String type) {
...
+    /**
+     * Finds the value of the given enumeration by name, case-insensitive.
+     * Throws an IllegalArgumentException if no match is found.
+     *
+     * @param entityType : string to compare against the value of EntityType
+     * @return EntityType : Object of type EntityType
+     * @throws IllegalArgumentException incase the argument is not present in the EntityType enum
+     **/
+
+    public static EntityType fromString(String entityType) {
```
Comment (pallavi-rao) [inline client/src/main/java/org/apache/falcon/entity/v0/EntityType.java]: There is already a getEnum method in EntityType, which converts a given string to uppercase and then validates.

As discussed, ideally, all the internal classes understand EntityType enum, if the DB requires it as a string, the Bean should handle internal conversion for read/write appropriately.

Comment (PraveenAdlakha) [inline client/src/main/java/org/apache/falcon/entity/v0/EntityType.java]: getEnum will work perfectly here. There is no need to make the changes we are writing to DB ,the purpose of this method was to introduce caseInsensitivity while getting the EnumType it can easily be served by getEnum.

File: client/src/main/java/org/apache/falcon/entity/v0/EntityType.java [hunk 2]
```diff
@@ -41,6 +41,7 @@
     PROCESS(Process.class, "/process-0.1.xsd", "name"),
     CLUSTER(Cluster.class, "/cluster-0.1.xsd", "name"),
     DATASOURCE(Datasource.class, "/datasource-0.1.xsd", "name");
+
```
Comment (pallavi-rao) [inline client/src/main/java/org/apache/falcon/entity/v0/EntityType.java]: Nit: Unnecessary changes empty line additions in this file.
